src/dune

```diff
@@ -8,39 +8,33 @@
   (modules (:standard \ main))
 )
 
+; HACK: workaround for https://github.com/ocaml/dune/issues/3374
+(rule (action (copy ../src__machdep-ml.c machdep-ml.c)))
+(rule (action (copy ../src__cilversion.ml cilversion.ml)))
+
 (rule
-  (targets machdep.ml cilversion.ml)
-  (deps ../configure.ac ../Makefile.in ../install-sh ../config.sub ../config.guess ../config.h.in ../stamp-h.in cil.mli machdep-ml.c.in cilversion.ml.in)
-  (action (chdir .. (progn (run ./configure) (run make machdep) (run cp _build/machdep.ml src))))
-)
+ (alias machdep)
+ (deps (alias ../configure) machdep-ml.c)
+ (targets machdep.ml)
+ (locks ../make)
+ (action (chdir .. (progn
+  (run make machdep)
+  (run cp _build/machdep.ml src)))))
 
 (rule
- (deps (package goblint-cil) ../src/main.exe (source_tree ../lib/perl5/App/Cilly) (source_tree ../test) ../bin/cilly ../configure.ac ../Makefile.in ../install-sh ../config.sub ../config.guess ../lib/perl5/App/Cilly.pm.in ../lib/perl5/Makefile.PL  ../config.h.in ../stamp-h.in cil.mli machdep-ml.c.in cilversion.ml.in ../config.mk.in)
- (action (chdir ..  (progn
-    (run ./configure)
-    (run make machdep)
-    (run make lib/perl5/App/Cilly.pm)
-    (chdir lib/perl5 (progn
-      (run perl Makefile.PL)
-      (run make)
-    ))
-    (run cp src/main.exe bin/cilly.native)
-    (bash "mkdir -p share/cil && echo $(cd ../install/default/lib; pwd) > share/cil/ocamlpath")
-    (run make test)
- )))
- (alias runtest)
-)
+ (alias cilly)
+ (deps (package goblint-cil))
+ (action (chdir .. (bash "mkdir -p share/cil && echo $(cd ../install/default/lib; pwd) > share/cil/ocamlpath")))) ; TODO: what is this?
```

Comment:
I have no idea what this is for. If it's redundant, then this entire rule can be removed since all other parts were extracted elsewhere.

Reply:
I think this was required for the regression tests to work, not sure if this still is the case...

Reply:
`CilCompiler.pm` from cilly seems to optionally read it. So maybe it'll only fail during actual cilly usage?

The tests run fine without that though, but they need the `(package goblint-cil)` dependency, for cilly to be installed into `_build/install/` for testing.